pkg/check/cluster.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/pingcap/log"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	pmysql "github.com/pingcap/tiflow/pkg/sink/mysql"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+var (
+	dbConnImpl  pmysql.IDBConnectionFactory = &pmysql.DBConnectionFactory{}
+	checkIsTiDB                             = pmysql.CheckIsTiDB
+	// GetClusterIDBySinkURIFn is a function to get the cluster ID by the sink URI.
+	// Exported for testing.
+	GetClusterIDBySinkURIFn = getClusterIDBySinkURI
+)
+
+// UpstreamDownstreamNotSame checks whether the upstream and downstream are not the same cluster.
+func UpstreamDownstreamNotSame(ctx context.Context, upPD pd.Client, downSinkURI string) (bool, error) {
+	upID := upPD.GetClusterID(ctx)
+
+	downID, isTiDB, err := GetClusterIDBySinkURIFn(ctx, downSinkURI)
+	log.Debug("CheckNotSameUpstreamDownstream",
+		zap.Uint64("upID", upID), zap.Uint64("downID", downID), zap.Bool("isTiDB", isTiDB))
+	if err != nil {
+		log.Error("failed to get cluster ID from sink URI",
+			zap.String("downSinkURI", downSinkURI), zap.Error(err))
+		return false, cerror.Trace(err)
+	}
+
+	if !isTiDB {
+		return true, nil
+	}
+
+	return upID != downID, nil
+}
+
+// getClusterIDBySinkURI gets the cluster ID by the sink URI.
+// Returns the cluster ID, whether it is a TiDB cluster, and an error.
+func getClusterIDBySinkURI(ctx context.Context, sinkURI string) (uint64, bool, error) {
```

Comment:
Let the mysql sink implement such functionality, and pass argument to the `UpstreamDownstreamNotSame` method

Reply:
Implementing it in here makes it eaiser to centralize testing for this entire functionality. It would still require importing mysql sink package if this function is implemented in there.